### ISSUE TITLE
An exception has been thrown during the rendering of a template ("Notice: Undefined index: _self")

### ISSUE THREAD

**Issue by arderyp** [twigphp/Twig#3055]
My symfony project is restricted to 3.4.x.

I ran composer update and my twig was updated from v2.6.2 => v2.11.2

After doing that, I started to receive some errors in macros where I was using the old approach mentioned here: ["Before Twig 2.11, when you want to use a macro in another macro from the same file, you need to import it locally..."](https://twig.symfony.com/doc/2.x/tags/macro.html)

So, I made changes like:

```
{% macro functionA(options) %}
  {% import _self as functions %}
  {{ functions.functionB(options) }}
{% endmacro functionA %}

{% macro functionB(options) %}
  ...
{% endmacro functionB %}
```

to

```
{% macro functionA(options) %}
  {{ _self.functionB(options) }}
{% endmacro functionA %}

{% macro functionB(options) %}
  ...
{% endmacro functionB %}
```

I made these changes based on the "macro scoping" documentation and examples [here](https://twig.symfony.com/doc/2.x/tags/macro.html). However, and I get the following error:

```
An exception has been thrown during the rendering of a template ("Notice: Undefined index: _self").
```

Is this a compatibility issue where twig 2.1.x is incompatible with symfony 3.4.x, or am I doing something wrong?  For what it is worth, I have a macro file with multiple macros, some of which call eachother.  Then I import the main function I want to call in an actual HTML template file, and execute it.  So, to follow the example above, my HTML template might include:

```
...
<div id="something">
  {% import 'functions.macro.twig' as macros %}
  {{ macros.functionA({
    'type' : 'table',
    'required': true,
    'container': false,
  }) }}
</div>
...
```

**Comment by fabpot** [comment]
I've just tested your example and it works for me. See https://twigfiddle.com/rfv32r

**Comment by arderyp** [comment]
Neat, I wasn't previously aware of twigfiddle.  I am trying to get an example that more accurately reflects the situation than your fiddle, but can't seem to make it work.  The import seems to be failing... care to have a look?

https://twigfiddle.com/ayhr05

**Comment by fabpot** [comment]
@arderyp You've just found a bug :) I will try to work on a fix later this week.

**Comment by arderyp** [comment]
Awesome!  Not awesome that this is a bug, but awesome that I am not losing my mind.  I really appreciate your help.

**Comment by fabpot** [comment]
@arderyp fixed in #3057

**Comment by arderyp** [comment]
awesome @fabpot, thanks!

**Comment by arderyp** [comment]
I just wanted to confirm that your fix (in v2.11.3) resolved my problem.  Thanks again!
